soh/soh/Enhancements/randomizer/randomizer.cpp

```diff
@@ -738,119 +738,141 @@ void Randomizer::ParseRandomizerSettingsFile(const char* spoilerFileName) {
                         if(it.value() == "Start With") {
-                            gSaveContext.randoSettings[index].value = 0;            
+                            gSaveContext.randoSettings[index].value = RO_DUNGEON_ITEM_LOC_STARTWITH;            
                         } else if(it.value() == "Vanilla") {
-                            gSaveContext.randoSettings[index].value = 1;
+                            gSaveContext.randoSettings[index].value = RO_DUNGEON_ITEM_LOC_VANILLA;
                         } else if(it.value() == "Own Dungeon") {
-                            gSaveContext.randoSettings[index].value = 2;
+                            gSaveContext.randoSettings[index].value = RO_DUNGEON_ITEM_LOC_OWN_DUNGEON;
                         } else if(it.value() == "Any Dungeon") {
-                            gSaveContext.randoSettings[index].value = 3;
+                            gSaveContext.randoSettings[index].value = RO_DUNGEON_ITEM_LOC_ANY_DUNGEON;
                         } else if(it.value() == "Overworld") {
-                            gSaveContext.randoSettings[index].value = 4;
+                            gSaveContext.randoSettings[index].value = RO_DUNGEON_ITEM_LOC_OVERWORLD;
                         } else if(it.value() == "Anywhere") {
-                            gSaveContext.randoSettings[index].value = 5;
+                            gSaveContext.randoSettings[index].value = RO_DUNGEON_ITEM_LOC_ANYWHERE;
                         }
                         break;
                     case RSK_GANONS_BOSS_KEY:
                         if(it.value() == "Vanilla") {
-                            gSaveContext.randoSettings[index].value = 0;
+                            gSaveContext.randoSettings[index].value = RO_GANON_BOSS_KEY_VANILLA;
                         } else if(it.value() == "Own dungeon") {
-                            gSaveContext.randoSettings[index].value = 1;
+                            gSaveContext.randoSettings[index].value = RO_GANON_BOSS_KEY_OWN_DUNGEON;
                         } else if(it.value() == "Start with") {
-                            gSaveContext.randoSettings[index].value = 2;
+                            gSaveContext.randoSettings[index].value = RO_GANON_BOSS_KEY_STARTWITH;
                         } else if(it.value() == "Any Dungeon") {
-                            gSaveContext.randoSettings[index].value = 3;
+                            gSaveContext.randoSettings[index].value = RO_GANON_BOSS_KEY_ANY_DUNGEON;
                         } else if(it.value() == "Overworld") {
-                            gSaveContext.randoSettings[index].value = 4;
-                        } else if(it.value() == "Anywhere") {
+                            gSaveContext.randoSettings[index].value = RO_GANON_BOSS_KEY_OVERWORLD;
+                        } else if(it.value() == "Anywhere") {                         
                             gSaveContext.randoSettings[index].value = 5;
```

Comment:
Looks like a couple of Ganon Boss Key options got added that don't have an enum yet. Mind adding those real quick before we merge this in?

Reply:
Done. Also added a few more enums for other settings that weren't caught either.